Fix channel results breaking Video search

FosterSamuel opened the pull request:
The video module should not abandon requests if a result is a channel link.

## Example Problem
If you say to JARVIS "linkin park videos", the last result is a channel link. This result causes JARVIS to abandon the search and default to the "I couldn't find any videos" message.

## Problem and Solution
Not all results from the YouTube API have `videoId` inside of `id`.

```python
item_url = 'https://www.youtube.com/watch?v=' + item['id']['videoId']
```

When the result is a channel link, it instead contains `channelId` inside of `id`. This pull request covers that reality with:
```python
try:
      # Video result from search
      item_url = 'https://www.youtube.com/watch?v=' + item['id']['videoId']
      buttons.add_web_url('YouTube Link', 'https://www.youtube.com/watch?v=' + item['id']['videoId'])
except:
      # Channel result from search
      item_url = 'https://www.youtube.com/channel/' + item['id']['channelId']      
```

Both results need a "Channel Link", but only one of them needs a "YouTube Link" button.

## Semantic Argument
Maybe a channel result when searching for videos is not right. The answer to this is to ignore channel results. If @swapagarwal wants that, I will go in that direction. 👍 

Regardless of direction, I do not believe the video module should give up on the search because of one channel result.

swapagarwal commented:
@FosterSamuel This is a really detailed write-up. Thanks for taking the time to deep-dive into the issue! 😄

swapagarwal commented on modules/src/video.py:
```diff
@@ -21,11 +21,16 @@ def process(input, entities):
+                buttons.add_web_url('YouTube Link', 'https://www.youtube.com/watch?v=' + item['id']['videoId'])
+            except:
+                # Channel result from search
+                item_url = 'https://www.youtube.com/channel/' + item['id']['channelId']
```
This might also result in an error, in case the result is of type `playlist`. See https://developers.google.com/youtube/v3/docs/search

FosterSamuel commented:
Gladly! 👍

Thank you for noticing the chance of a playlist result. I'll work on adding that soon! 

_Sent from my Google Nexus 6P using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

FosterSamuel commented:
Gladly! 👍

Thank you for noticing the chance of a playlist result. I'll work on adding that soon!  

_Sent from my Google Nexus 6P using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

FosterSamuel commented:
Gladly! 👍

Thank you for noticing the chance of a playlist result. I'll work on adding that soon!

FosterSamuel commented:
# Changes 
1. Removed `type=video` from API query to allow Playlist results 
2. Changed "YouTube Link" --> "Video Link" for clarity
3. Added code to handle all three types of results

swapagarwal commented:
If `type` was set to `video` before, how come `channel` was returned in response? 🤔

FosterSamuel commented:
@swapagarwal Not sure. Seems to be a less strict part of the YouTube API. 

If you use the first commit (with `type=video` in the query), it exposes the results of a search that may have channel results in it. The type only excluded playlists, for some reason. 

I can record a video example of this if you'd like further proof of the statements.
